tensilelite/Tensile/Source/client/include/DataInitialization.hpp

```diff
@@ -316,7 +318,7 @@ namespace TensileLite
                 }
 
                 if(m_gpuInit && m_curBoundsCheck == BoundsCheckMode::Disable
-                   && !m_problemDependentData)
+                   && !m_problemDependentData && !needSwizzle)
                 {
                     if(m_elementsToValidate)
```

Comment:
If needSwizzle, then we can't just do resetOutput. We still need to do a swizzling and then copy to GPU for this problem.